common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -30,7 +30,18 @@ case class ContainerArgsConfig(network: String,
                                dnsServers: Seq[String] = Seq.empty,
                                dnsSearch: Seq[String] = Seq.empty,
                                dnsOptions: Seq[String] = Seq.empty,
-                               extraArgs: Map[String, Set[String]] = Map.empty)
+                               extraEnvVars: Seq[String] = Seq.empty,
+                               extraArgs: Map[String, Set[String]] = Map.empty) {
+
+  val extraEnvVarMap: Map[String, String] =
+    extraEnvVars.flatMap {
+      _.split("=", 2) match {
+        case Array(key)        => Some(key -> "")
+        case Array(key, value) => Some(key -> value)
+        case _                 => None
```

Comment:
I think the _ case is unreachable and maybe we don't need the Option type now since a failure to split on `=` can be interpreted as a request to simply set an envvar with an empty value.  But I don't think it hurts to do it this way.